blockball-bukkit-plugin/src/main/java/com/github/shynixn/blockball/bukkit/logic/business/service/ParticleServiceImpl.kt

```diff
@@ -57,10 +59,15 @@ class ParticleServiceImpl @Inject constructor(
      * Plays the given [particle] at the given [location] for the given [players].
      */
     override fun <L, P> playParticle(location: L, particle: Particle, players: Collection<P>) {
-        if (location !is Location) {
-            throw IllegalArgumentException("Location has to be a BukkitLocation!")
+        if (location is Position) {
+            playParticle(location.toLocation(), particle, players)
+        }else if (location is Location) {
+            playParticle(location, particle, players)
         }
+        throw IllegalArgumentException("Location has to be a BukkitLocation!")
```

Comment:
This method always throws an IllegalArgumentException when called. This needs to get fixed.